src/ocl/convolutionocl.cpp

```diff
@@ -402,6 +377,84 @@ static void ConvForwardCheckNumerics(const Handle& handle,
     }
 }
 
+void ConvolutionDescriptor::ValidateConvTensors(const ConvTensors& tensors) const
+{
+
+    // Group stride in current TensorDescriptor is implicit. When invoking kernels,
+    // we need to add the group dimension G and compute its stride. We want the stride
+    // left of C to be a multiple of group count G. e.g. for NCHW, the stride for N
+    // should be a multiple of G so that we can compute the strides for NGCHW
```

Comment:
Here the functionality overlaps with ValidateGroupCount(). We also have Problem::ValidateGroupCount(). It would be nice to sort this out.

Reply:
https://github.com/ROCmSoftwarePlatform/MIOpen/issues/2444 created to handle this. 

Proposal: We remove `Problem::ValidateGroupCount()` and then combine `ValidateGroupCount()` in `convolutionocl` with `conv::ProblemDescription::ValidateTensors()` (currently `ConvDescriptor::ValidateTensors()` and use it everywhere. This will allow getting `N` and `C` lengths and strides easily from accessors in `conv::ProblemDescrioption`. 

Alternate Proposal: Combine all `ValidateGroupCount` with `ConvDescriptor::ValidateTensors()`. Downside: this requires logic to extract `N` and `C` lengths and strides.